backend/src/Designer/Services/Implementation/TextResourceService.cs

```diff
@@ -41,68 +41,40 @@ public class TextResourceService : ITextResourceService
         }
 
         /// <inheritdoc/>
-        public async Task UpdateTextResourcesAsync(string org, string app, string shortCommitId, string envName)
+        public async Task UpdateTextResourcesAsync(string org, string app, string shortCommitId, string envName, CancellationToken cancellationToken = default)
         {
+            cancellationToken.ThrowIfCancellationRequested();
             string textResourcesPath = GetTextResourceDirectoryPath();
             List<FileSystemObject> folder = await _giteaApiWrapper.GetDirectoryAsync(org, app, textResourcesPath, shortCommitId);
-            if (folder != null)
+            if (folder == null)
             {
-                // TODO: Add Cancellation token to signature of the method and to the methods used in loop and convert loot to Parallel.ForEachAsync
-                // https://github.com/Altinn/altinn-studio/issues/12031
-                foreach (FileSystemObject textResourceFromRepo in folder)
-                {
-                    if (!Regex.Match(textResourceFromRepo.Name, "^(resource\\.)..(\\.json)").Success)
-                    {
-                        continue;
-                    }
+                return;
+            }
```

Comment:
I love this change! It significantly enhances readability, eliminating the need to verify the existence of an 'else' clause after the 'if' statement. This approach assures me that no further operations occur below within this method. 👏